tests/accessor/local_accessor_constructors.h

```diff
@@ -0,0 +1,118 @@
+#define SYCL_CTS_LOCAL_ACCESSOR_H
+#include "catch2/catch_test_macros.hpp"
+
+#include "accessor_common.h"
+
+namespace local_accessor_constructors {
+using namespace sycl_cts;
+using namespace accessor_tests_common;
+
+constexpr accessor_type AccTypeT = accessor_type::local_accessor;
+
+template <typename DataT, int DimensionT>
+void test_default_constructor(const std::string& type_name) {
+  const auto section_name =
+      get_section_name<DimensionT>(type_name, "Default constructor");
+
+  SECTION(section_name) {
+    auto get_acc_functor = [] {
+      return sycl::local_accessor<DataT, DimensionT>();
+    };
+    if constexpr (std::is_const_v<DataT>) {
+      check_def_constructor<AccTypeT, DataT, DimensionT,
+                            sycl::access_mode::read, sycl::target::device>(
+          get_acc_functor);
+    } else {
+      check_def_constructor<AccTypeT, DataT, DimensionT,
+                            sycl::access_mode::read_write,
+                            sycl::target::device>(get_acc_functor);
+    }
+  }
+}
+
+template <typename DataT>
+void test_zero_dimension_buffer_constructor(const std::string& type_name) {
+  const auto section_name =
+      get_section_name<0>(type_name, "Zero dimension constructor");
+
+  SECTION(section_name) {
+    auto get_acc_functor = [](sycl::buffer<DataT, 1>& data_buf,
+                              sycl::handler& cgh) {
+      return sycl::local_accessor<DataT, 0>(cgh);
+    };
+    if constexpr (std::is_const_v<DataT>) {
+      check_zero_dim_constructor<AccTypeT, DataT, sycl::access_mode::read,
+                                 sycl::target::device>(get_acc_functor);
+    } else {
+      check_zero_dim_constructor<AccTypeT, DataT, sycl::access_mode::read_write,
+                                 sycl::target::device>(get_acc_functor);
+    }
+  }
+}
+
+template <typename DataT, int DimensionT>
+void test_common_constructors(const std::string& type_name) {
+  auto r = util::get_cts_object::range<DimensionT>::get(1, 1, 1);
+  auto offset = sycl::id<DimensionT>();
+
+  auto section_name =
+      get_section_name<DimensionT>(type_name, "From sycl::range constructor");
+
+  SECTION(section_name) {
+    auto get_acc_functor = [r](sycl::buffer<DataT, DimensionT>& data_buf,
+                               sycl::handler& cgh) {
+      return sycl::local_accessor<DataT, DimensionT>(r, cgh);
+    };
+    if constexpr (std::is_const_v<DataT>) {
+      check_common_constructor<AccTypeT, DataT, DimensionT,
+                               sycl::access_mode::read, sycl::target::device>(
+          get_acc_functor);
+    } else {
+      check_common_constructor<AccTypeT, DataT, DimensionT,
+                               sycl::access_mode::read_write,
+                               sycl::target::device>(get_acc_functor);
+    }
+  }
+}
+
+template <typename T, typename DimensionTypeT>
+class run_tests_constructors {
+  static constexpr int DimensionT = DimensionTypeT::value;
```

Comment:
```suggestion
  static constexpr int Dimension = DimensionT::value;
```
Nit: it might be a bit confusing to use a `T` suffix for non-type template parameters, what do you think?

Reply:
I did it purpose. I wanted to make `DimensionT` looks like a template parameter, because `DimensionT` 100% depends on `typename DimensionTypeT` which is an actual template parameter. It's kind of workaround.

We can't get rid of `typename DimensionTypeT` completely and change it to `size_t DimensionT` because of limitation of `for_all_combination` (at least at this point I don't know how to do it).

Reply:
AFAIK the `T` suffix usually stands for type, not for template parameter - see `std::make_signed_t` for example with `snake_case` naming. My concern is that `DimensionT` looks like type, not like an integral value.

Reply:
Good point. 
I also checked paragraph 4.7.6.9.1. of the spec, to see naming of template parameters. It appears that accessor in spec doesn't have `T` suffix in `Dimensions` or `Target` template parameters. I will fix this issue in all accessor PRs.

Reply:
Done.

Reply:
Great! Haven't thought about SYCL 2020 spec, thanks for in-depth background check :)

Reply:
Yes, the template arguments have been cleaned up relatively recently https://github.com/KhronosGroup/SYCL-Docs/pull/147